compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirMemberPropertyChecker.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.descriptors.ClassKind
+import org.jetbrains.kotlin.descriptors.Visibilities
+import org.jetbrains.kotlin.fir.FirSourceElement
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.getModifierList
+import org.jetbrains.kotlin.fir.analysis.checkers.source
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.fir.declarations.impl.FirDefaultPropertyAccessor
+import org.jetbrains.kotlin.fir.expressions.FirExpression
+import org.jetbrains.kotlin.fir.symbols.impl.FirPropertyAccessorSymbol
+import org.jetbrains.kotlin.fir.types.FirImplicitTypeRef
+import org.jetbrains.kotlin.lexer.KtTokens.ABSTRACT_KEYWORD
+import org.jetbrains.kotlin.lexer.KtTokens.PRIVATE_KEYWORD
+
+// See old FE's [DeclarationsChecker]
+object FirMemberPropertyChecker : FirBasicDeclarationChecker() {
+    override fun check(declaration: FirDeclaration, context: CheckerContext, reporter: DiagnosticReporter) {
+        if (declaration !is FirRegularClass) {
+            return
+        }
+        for (member in declaration.declarations) {
+            if (member is FirProperty) {
+                checkProperty(declaration, member, reporter)
+            }
+        }
+    }
+
+    private fun checkProperty(containingDeclaration: FirRegularClass, property: FirProperty, reporter: DiagnosticReporter) {
+        if (inInterface(containingDeclaration) &&
+            property.visibility == Visibilities.Private &&
+            !property.isAbstract &&
+            (property.getter == null || property.getter is FirDefaultPropertyAccessor)
+        ) {
+            property.source?.let { source ->
+                val modifier = source.getModifierList()?.modifiers?.find { it.token == PRIVATE_KEYWORD }
```

Comment:
Please don't use `FirModifierList`, it's a specific class created for the purposes of `FirModifierChecker` (recently I moved it inside to avoid such misunderstandings). To report on modifier, use positioning strategies instead. See `RedundantModalityModifierChecker` as an example.

Reply:
Also, you can use various helpers from `LightTreePositioningStrategies.kt` to check existence of some modifier or child.

Reply:
Thanks for the pointer. With positioning strategies, not needed to use helpers at all, cool!